controllers/osccluster_routetable_controller_unit_test.go

```diff
@@ -0,0 +1,2774 @@
+			internetServiceRef.ResourceMap = make(map[string]string)
+			internetServiceRef.ResourceMap[internetServiceName] = internetServiceId
+
+			natServiceName := rttc.spec.Network.NatService.Name + "-uid"
+			natServiceId := "nat-" + natServiceName
+			natServiceRef := clusterScope.GetNatServiceRef()
+			natServiceRef.ResourceMap = make(map[string]string)
+
+			routeRef := clusterScope.GetRouteRef()
+			routeRef.ResourceMap = make(map[string]string)
+
+			linkRouteTableRef := clusterScope.GetLinkRouteTablesRef()
+			linkRouteTableRef.ResourceMap = make(map[string]string)
+
+			routeTablesRef := clusterScope.GetRouteTablesRef()
+			routeTablesRef.ResourceMap = make(map[string]string)
+
+			var associateRouteTableId string
+			var resourceId string
+			routeTablesSpec := rttc.spec.Network.RouteTables
+			for _, routeTableSpec := range routeTablesSpec {
+				routeTableName := routeTableSpec.Name + "-uid"
+				routeTableId := "rtb-" + routeTableName
+				linkRouteTableId := "eipalloc-" + routeTableName
+				routeTablesRef.ResourceMap[routeTableName] = routeTableId
+				associateRouteTableId = routeTableId
+
+				linkRouteTableRef.ResourceMap[routeTableName] = linkRouteTableId
+
+				mockOscRouteTableInterface.
+					EXPECT().
+					GetRouteTableIdsFromNetIds(gomock.Eq(netId)).
+					Return([]string{routeTableId}, rttc.expGetRouteTableIdsFromNetIdsErr)
+
+				routesSpec := routeTableSpec.Routes
+				for _, routeSpec := range routesSpec {
+					destinationIpRange := routeSpec.Destination
+					resourceType := routeSpec.TargetType
+					routeName := routeSpec.Name + "-uid"
+					routeRef.ResourceMap[routeName] = routeTableId
+					if resourceType == "gateway" {
+						resourceId = internetServiceId
+					} else {
+						resourceId = natServiceId
+					}
+					route := osc.CreateRouteResponse{
+						RouteTable: &osc.RouteTable{
+							RouteTableId: &routeTableId,
+						},
+					}
+
+					readRouteTables := osc.ReadRouteTablesResponse{
+						RouteTables: &[]osc.RouteTable{
+							*route.RouteTable,
+						},
+					}
+
+					readRouteTable := *readRouteTables.RouteTables
+					mockOscRouteTableInterface.
+						EXPECT().
+						GetRouteTableFromRoute(gomock.Eq(associateRouteTableId), gomock.Eq(resourceId), gomock.Eq(resourceType)).
+						Return(&readRouteTable[0], rttc.expGetRouteTableFromRouteErr)
+					mockOscRouteTableInterface.
+						EXPECT().
+						DeleteRoute(destinationIpRange, routeTableId).
+						Return(rttc.expDeleteRouteErr)
+				}
+			}
+			reconcileDeleteRouteTable, err := reconcileDeleteRouteTable(ctx, clusterScope, mockOscRouteTableInterface)
+			if err != nil {
+				assert.Equal(t, rttc.expReconcileDeleteRouteTableErr.Error(), err.Error(), "reconcileDeleteRouteTable() should return the same error")
+			} else {
+				if rttc.expReconcileDeleteRouteTableErr != nil {
+					t.Errorf("reconcileDeleteRouteTable() expected error = %s but get no error", rttc.expReconcileDeleteRouteTableErr.Error())
+				}
```

Comment:
It is just a detail, no need to update. You could have an assert here too for the type.